### ISSUE TITLE
Initialisation of static members broken after upgrading from v15.0.4 to v15.1.0

### ISSUE THREAD

**Issue by heinerlamprecht** [angular/angular#48764]
### Which @angular/* package(s) are the source of the bug?

compiler, core

### Is this a regression?

Yes

### Description

After upgrading from v15.0.4 to v15.1.0, I do get compile errors for code like this:

`
@Injectable
export class MyService {
    private static MY_URL_SCHEME = 'my app://';
    private static MY_URL_SCHEME_LENGTH = MyService.MY_URL_SCHEME.length;
}
`

Error message during 'ng test':

`TypeError: Cannot read properties of undefined (reading 'CUSTOM_URL_SCHEME')`

### Please provide the environment you discovered this bug in (run `ng version`)

Angular CLI: 15.1.1
Node: 16.18.1
Package Manager: npm 8.19.2
OS: darwin arm64

Angular: 15.1.0
... animations, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, router, service-worker

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1501.1
@angular-devkit/build-angular   15.1.1
@angular-devkit/core            15.1.1
@angular-devkit/schematics      15.1.1
@angular/cli                    15.1.1
@schematics/angular             15.1.1
rxjs                            6.6.7
typescript                      4.8.4

**Comment by alan-agius4** [comment]
The problem appears to have been introduced by https://github.com/angular/angular/pull/48638 which was first available on NPM in `15.1.0-rc.0`.

//cc @crisbeto

**Comment by Bil0** [comment]
Hello, the problem appears to be only in `serve mode` and not in `prod mode`

**Comment by JoostK** [comment]
tldr; this is caused by microsoft/typescript#52004.

---

This reproduces when `"module": "es2022"` is configured in tsconfig.json, but _not_ for `es2015`.

Under `ng test` with `es2022`, this is the (broken) output:

```js
let MyService = MyService_1 = (_class2 = class MyService {},
  (0,_Users_joost_Development_Github_angular_repros_ng24560_node_modules_babel_runtime_helpers_esm_defineProperty_js__WEBPACK_IMPORTED_MODULE_0__["default"])(_class2, "MY_URL_SCHEME", 'my app://'),
  (0,_Users_joost_Development_Github_angular_repros_ng24560_node_modules_babel_runtime_helpers_esm_defineProperty_js__WEBPACK_IMPORTED_MODULE_0__["default"])(_class2, "MY_URL_SCHEME_LENGTH", MyService_1.MY_URL_SCHEME.length),
_class2);
MyService = MyService_1 = (0,tslib__WEBPACK_IMPORTED_MODULE_4__.__decorate)([(0,_angular_core__WEBPACK_IMPORTED_MODULE_5__.Injectable)({
  providedIn: 'root'
})], MyService);
```

When using `es2015`, the following is emitted:

```js
let MyService = MyService_1 = class MyService {};
Object.defineProperty(MyService, "MY_URL_SCHEME", {
  enumerable: true,
  configurable: true,
  writable: true,
  value: 'my app://'
});
Object.defineProperty(MyService, "MY_URL_SCHEME_LENGTH", {
  enumerable: true,
  configurable: true,
  writable: true,
  value: MyService_1.MY_URL_SCHEME.length
});
MyService = MyService_1 = (0,tslib__WEBPACK_IMPORTED_MODULE_3__.__decorate)([(0,_angular_core__WEBPACK_IMPORTED_MODULE_4__.Injectable)({
  providedIn: 'root'
})], MyService);
```

This behavior corresponds with TypeScript bug microsoft/typescript#52004.

For completeness, the code that was generated before #48638 was

```js
let MyService = (_class2 = class MyService {},
  (0,_Users_joost_Development_Github_angular_repros_ng24560_node_modules_babel_runtime_helpers_esm_defineProperty_js__WEBPACK_IMPORTED_MODULE_0__["default"])(_class2, "MY_URL_SCHEME", 'my app://'),
  (0,_Users_joost_Development_Github_angular_repros_ng24560_node_modules_babel_runtime_helpers_esm_defineProperty_js__WEBPACK_IMPORTED_MODULE_0__["default"])(_class2, "MY_URL_SCHEME_LENGTH", _class2.MY_URL_SCHEME.length),
_class2);
MyService = (0,tslib__WEBPACK_IMPORTED_MODULE_4__.__decorate)([(0,_angular_core__WEBPACK_IMPORTED_MODULE_5__.Injectable)({
  providedIn: 'root'
})], MyService);
```

i.e. the alias was avoided entirely, avoiding the TypeScript bug.

I would qualify this as upstream issue; the change in #48638 did indeed affect how this behaves but that change was needed to fix another issue and it was effectively a revert to how it used to work before 15.0. I suspect that pre-15.0 the application was not using `"module": "es2022"`, hence the bug only became apparent in 15.1.

**Comment by andreas-aeschlimann** [comment]
I ran into this when upgrading our application from Angular 14 to Angular 15. There are no build errors on dev or prod, but with `ng serve` the application won't run. Changing the target to `ES2021` was a workaround for us.

Is this an issue coming from TS that will be fixed there at some point?
